### PR TITLE
Bvh accel

### PR DESCRIPTION
**Pull request by madmann91**
This PR implements a binned BVH builder. In its current state, the code needs benchmarking to figure out if things are faster, or slower. I cannot really do that as RVPT is only giving a couple of frames a second on my machine, in either version (i.e. before or after the changes).

### CODE REVIEW

File: src/rvpt/bvh.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿//
```
Comment (LegendWasTaken) [inline src/rvpt/bvh.cpp:1]: I should really go and remove these.

Comment (madmann91) [reply src/rvpt/bvh.cpp:1]: Eh, it's technically correct so it's fine.

File: src/rvpt/rvpt.h [hunk 2]
```diff
@@ -112,22 +112,15 @@ class RVPT
...
+
+    // Debug BVH view
     std::vector<std::vector<AABB>> depth_bvh_bounds;
     size_t bvh_vertex_count = 0;
-    int max_bvh_build_depth = 5;
-    int max_bvh_view_depth = 2;
+    int max_bvh_view_depth = 1;
     bool view_previous_depths = true;
-    bool does_bvh_support_depth_viewing = false;
-    // Yes I know, I have an extra set of the AABB's, I did this so instead of recursively going
```
Comment (madmann91) [inline src/rvpt/rvpt.h:123]: I took the liberty of removing that comment. You can add it back if you want. I think the current solution of having a copy of the AABBs makes sense. If you don't like that you can alternatively use `bvh::collect_aabbs_by_depth()` to reconstruct them.

Comment (LegendWasTaken) [reply src/rvpt/rvpt.h:123]: Oh yeah it's fine, 99% of this code written by me is really old and at this point I would write it completely differently anyways.

File: src/rvpt/camera.cpp [hunk 3]
```diff
@@ -57,6 +57,9 @@ void Camera::recalculate_values()
     float view_height = tanf(theta / 2);
     float view_width = aspect * view_height;
 
+    translation = glm::vec3(0.784, 1.631, -0.564);
+    rotation = glm::vec3(-82.20, 42, 0);
+
```
Comment (LegendWasTaken) [inline src/rvpt/camera.cpp:62]: Is this just to make the camera not start in a weird place?

**Review by LegendWasTaken — APPROVED** [review]
Looks fine to me, Glad to have someone a lot more knowledgable about this handling it!

**Comment by LegendWasTaken** [comment]
Whats interesting is I was going to redo the entire BVH system myself, really glad (as I said in the review) that someone better than me went ahead and did it! Go open source

File: src/rvpt/camera.cpp (hunk 3 above)
Comment (madmann91) [reply src/rvpt/camera.cpp]: Should be gone now, I peeled the last commit and fixed it.

**Comment by LegendWasTaken** [comment]
Alright well, after over 8 days of no activity and a good pull request - Merging

**Pull request merged**